UnityMirror/UnityServer/Assets/Server/Scripts/AgentListener.cs

```diff
@@ -25,16 +25,23 @@ public class AgentListener : MonoBehaviour {
         UnityNetworkServer.Instance.OnPlayerRemoved.AddListener(OnPlayerRemoved);
 
         // get the port that the server will listen to
-        // we are getting the port via GSDK, the value of the "ListeningPortKey" must be the same value as the one you use when you create the Build
-        // or the one you use on LocalMultiplayerAgent JSON configuration file
         // We *have to* do it on process mode, since there might be more than one game server instances on the same VM and we want to avoid port collision
         // On container mode, we can omit the below code and set the port directly, since each game server instance will run on its own network namespace. However, below code will work as well
         // we have to do that on process
-        var config = PlayFabMultiplayerAgentAPI.GetConfigSettings();
-        if(config.ContainsKey(ListeningPortKey))
+        var connInfo = PlayFabMultiplayerAgentAPI.GetGameServerConnectionInfo();
+        // make sure the ListeningPortKey is the same as the one configured in your Build settings (either on LocalMultiplayerAgent or on MPS)
+        const string ListeningPortKey = "game_port";
```

Comment:
I have a preference towards making this variable static const ... and put it outside the class, and rename it as LISTENING_PORT_KEY, but probably this is too much influence by C++ standards.

Reply:
I disagree. It only needs to be static if it's public and used outside the class.
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constants
_Constants are accessed as if they were [static](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/static) fields because the value of the constant is the same for all instances of the type. You do not use the static keyword to declare them._

Reply:
You are right, in this case this value is only relevant for a function.

Reply:
thanks both. The const declaration was on top of the class but I moved it here so it's more visible for the end user convenience and ability to customize.